rusoto/core/src/signature.rs

```diff
@@ -380,7 +380,7 @@ impl SignedRequest {
                                             canonical_headers,
                                             signed_headers,
                                             &to_hexdigest(""));
-                (Some(to_hexdigest("")), None)
+                (Some(to_hexdigest("")), Some(0))
```

Comment:
This is the only part of the PR I'm not certain on. Does AWS want the content length header to be present even if there isn't anything? I don't know if our current behavior of omitting it is right or not. 🤔

Reply:
This was actually put in place because it's the reason Ceph was failing in SemaphoreCI. I can't imagine it would cause issues with AWS (after all, sending `0` is "correct" if there's no body).

Reply:
Huh, the [S3 docs](https://docs.aws.amazon.com/AmazonS3/latest/API/RESTCommonRequestHeaders.html) claim this header is required. Probably best we send something all the time, then. 😄